See Travis when I force a test to fail:
```python
=================================== FAILURES ===================================
_________________ MgmtEventGridTest.test_eventgrid_user_topics _________________
azure-sdk-testutils/devtools_testutils/mgmt_testcase.py:135: in tearDown
    return super(AzureMgmtTestCase, self).tearDown()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <test_azure_mgmt_eventgrid.MgmtEventGridTest testMethod=test_eventgrid_user_topics>
    def tearDown(self):
        os.environ = self.original_env
>       assert not [t for t in threading.enumerate() if t.name.startswith("AzureOperationPoller")]
E       AssertionError
../../../virtualenv/python3.4.6/lib/python3.4/site-packages/azure_devtools/scenario_tests/base.py:144: AssertionError
```

--- a/src/azure_devtools/scenario_tests/base.py
+++ b/src/azure_devtools/scenario_tests/base.py
@@ -141,7 +141,8 @@ class ReplayableTest(IntegrationTestBase):  # pylint: disable=too-many-instance-
 
     def tearDown(self):
         os.environ = self.original_env
-        assert not [t for t in threading.enumerate() if t.name.startswith("AzureOperationPoller")]
+        assert not [t for t in threading.enumerate() if t.name.startswith("AzureOperationPoller")], \
+            "You need to call 'result' or 'wait' on all AzureOperationPoller you have created"
 
     def _process_request_recording(self, request):
         if self.disable_recording:

